### DISCUSSION TITLE
Examples of GoogleDriveImport with JS Node client?

### DISCUSSION THREAD

**Discussion by impactcolor** [FlatIO/api-client-js#148] [Q&A] [answered]
Any examples? I see the ScoreCreation but not using the google drive import.
Thank you in advance!

**Comment by gierschv** [comment, marked as answer]
Hey there,

We have a request sample [in our REST API Docs](https://flat.io/developers/api/reference/#tag/Score/operation/createScore)

![Screenshot 2023-08-25 at 14 12 45](https://github.com/FlatIO/api-client-js/assets/396537/d5a1312f-af9b-479e-8f90-0f4aa4c77ffe)

Are you using it with this JS SDK? We haven't updated it for a while, I will need to check what OpenAPI Generator we would use these days.

**Reply by impactcolor** [reply]
@gierschv Yes I am using it with the JS SDK.
I had attempted that example with:

```
var body = new FlatApi.ScoreCreation(); // ScoreCreation | 

body = {
    "title": "Violin and Piano",
    "privacy": "private",
    "source": {
        "googleDrive": "1zoXasdfasdfQvuf1WkIujIFGgAJAD0E",
        "privacy": "private"
    }
  }
```

The error I get back is:       status: 500,
      text: '{"message":"An error occurred when processing the request, please retry later or contact our support at hello@flat.io","code":"INTERNAL_ERROR"}',
      method: 'POST',
      path: '/v2/scores'

**Reply by gierschv** [reply]
I think the problem here is that this file cannot be loaded using the file ID publicly (https://drive.google.com/file/d/1zoXasdfasdfQvuf1WkIujIFGgAJAD0E/view), and our Drive App has not be used with that file yet, so it doesn't have access rights on that.

I will check why this case returns a 500 error.

**Reply by impactcolor** [reply]
That was a dud file thou here goes a real one.  But the real one gave me the same error.  Just in case this throws you off.
https://drive.google.com/file/d/1zoXkVJnR-WPDQvuf1WkIujIFGgAJAD0E/view?usp=drive_link

**Reply by gierschv** [reply]
Can you try with a file with "Anyone with the link" sharing option in Drive? I don't know if there is a method otherwise in Drive to share a file with a specific app (that's what is done by Drive when you open a file with our app in drive).

**Reply by gierschv** [reply]
We just released some update on our Google Drive integration for imports:

- Fixed the imports of public files ("Anyone with link") - I bumped an issue on Google Issues Tracker for that one, we get some 404 form Drive when fetching public files on behalf of any user account.
- And now returning the drive errors instead of 500

Success:

```bash
curl 'https://api.flat.io/v2/scores' \ 
  -H 'content-type: application/json' \
  -H 'Authorization: Bearer xxx' \
  --data-raw '{"source": {"googleDrive": "1oAcNDuURlBeHHc8CdVsyNT0UrjA4snAp"}}'
{"id":"64e87a89cf70902e24c175d5","title":"Piano.musicxml"....
```

Drive error:

```bash
  curl 'https://api.flat.io/v2/scores' \
  -H 'content-type: application/json' \
  -H 'Authorization: Bearer xxx' \
  --data-raw '{"source": {"googleDrive": "1o"}}'
{"message":"[Google Drive Error] File not found: 1o.","code":"GOOGLE_DRIVE_ERROR"}
```
